Zend/zend_vm_def.h

```diff
@@ -4051,6 +4051,14 @@ ZEND_VM_HOT_HANDLER(130, ZEND_DO_UCALL, ANY, ANY, SPEC(RETVAL,OBSERVER))
 	SAVE_OPLINE();
 	EX(call) = call->prev_execute_data;
 
+	if (UNEXPECTED((fbc->common.fn_flags & ZEND_ACC_DEPRECATED) != 0)) {
```

Comment:
The changes in `ZEND_DO_UCALL` can probably be avoided by compiling to `ZEND_DO_FCALL` or `ZEND_DO_FCALL_BY_NAME` when the function is deprecated instead.

Reply:
Indeed, there's already a check for the deprecation flag in `zend_get_call_op()`:

https://github.com/php/php-src/blob/6c9d443a21062ad2d2c03c0f482976d29e6f285c/Zend/zend_compile.c#L3874

I'm wondering, though: Is the `ZEND_DO_FCALL_BY_NAME` for internal functions the correct choice or is this a pre-existing issue? My understanding is that it should `ZEND_DO_FCALL`, as the `init_op->opcode` is not `ZEND_INIT_FCALL_BY_NAME`?

Reply:
As mentioned in private, `ZEND_DO_UCALL` is a specialized version of `ZEND_DO_FCALL_BY_NAME`, so rather than adding more cases to handle in the specialized version, we can compile to the generic one.

Reply:
In case it wasn't clear, I'd also change the corresponding INIT calls. This case is relatively rare (comp-time known call to a deprecated user function).